ie/tsn-bridge-id.go

```diff
@@ -39,13 +39,17 @@ func (i *IE) TSNBridgeID() (net.HardwareAddr, error) {
 
 	switch i.Type {
 	case TSNBridgeID:
-		if has1stBit(i.Payload[0]) {
+		if !has1stBit(i.Payload[0]) {
+			return nil, nil
+		}
+
+		if len(i.Payload) < 9 {
 			if len(i.Payload) < 7 {
 				return nil, io.ErrUnexpectedEOF
 			}
 			return net.HardwareAddr(i.Payload[1:7]), nil
 		}
-		return nil, nil
+		return net.HardwareAddr(i.Payload[1:9]), nil
```

Comment:
I would suggest the following instead, to be able to also raise an error on malformed 8-bytes length IEs:
```golang
case TSNBridgeID:
	if !has1stBit(i.Payload[0]) {
		return nil, nil
	}
	// If the first bit is set to 1,
	// the payload contains
	// 1. a first byte, and either
	// 2a. exactly 6 bytes, or
	// 2b. at least 8 bytes.
	if (len(i.Payload) != 7) && (len(i.Payload) < 9) {
		return nil, ie.ErrUnexpectedEOF
	}
	// In 129.244 V16.2.0 to V16.3.1,
	// MAC address shall be encoded
	// as 12-digit hexadecimal numbers
	if len(i.Payload) == 7 {
		return net.HardwareAddr(i.Payload[1:7]), nil)
	}
	// Starting from V16.4.0, 
	// the User Plane Node Value, shall be encoded
	// as an IEEE 801Q Uint64
	return net.HardwareAddr(i.Payload[1:9]), nil)
```

Reply:
that's neat, would take it! @eric5800602 could you update?

Reply:
@wmnsk Sure.
Thanks for the suggestion.